File: generate_mock.js
const fs = require('fs');

const teams = [
  'USA', 'Mexico', 'Canada', 'Argentina', 'Brazil', 'France', 'England', 'Spain',
  'Germany', 'Portugal', 'Italy', 'Netherlands', 'Belgium', 'Croatia', 'Uruguay', 'Colombia',
  'Japan', 'South Korea', 'Iran', 'Australia', 'Senegal', 'Morocco', 'Nigeria', 'Egypt',
  'Saudi Arabia', 'Qatar', 'Ecuador', 'Peru', 'Chile', 'Sweden', 'Switzerland', 'Denmark',
  'Serbia', 'Poland', 'Wales', 'Scotland', 'Ukraine', 'Turkey', 'Hungary', 'Austria',
  'Czechia', 'Greece', 'Ivory Coast', 'Ghana', 'Cameroon', 'Mali', 'Algeria', 'Tunisia'
];

// Re-arrange to force Group D: USA, Australia, Paraguay, Turkey
// And maybe some interesting teams in Group A, etc.
const customGroups = {
  "GROUP_A": ["Brazil", "Scotland", "Ecuador", "Japan"],
  "GROUP_B": ["England", "Iran", "USA", "Wales"], // Randomly keeping some
  "GROUP_C": ["Argentina", "Saudi Arabia", "Mexico", "Poland"],
  "GROUP_D": ["USA", "Australia", "Paraguay", "Turkey"], // From the user's image! (Note ABD = USA, Avustralya = Australia, Paraguay, Turkiye)
  "GROUP_E": ["Spain", "Costa Rica", "Germany", "Japan"],
  "GROUP_F": ["Belgium", "Canada", "Morocco", "Croatia"],
  "GROUP_G": ["Brazil", "Serbia", "Switzerland", "Cameroon"],
  "GROUP_H": ["Portugal", "Ghana", "Uruguay", "South Korea"],
  "GROUP_I": ["Italy", "Nigeria", "Colombia", "Sweden"],
  "GROUP_J": ["Netherlands", "Senegal", "Chile", "Austria"],
  "GROUP_K": ["France", "Peru", "Denmark", "Tunisia"],
  "GROUP_L": ["Mexico", "Mali", "Greece", "Algeria"],
};

const standings = [];
const letters = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L"];

for (let i = 0; i < 12; i++) {
  const groupName = `GROUP_${letters[i]}`;
  const groupTeams = customGroups[groupName];
  
  const table = groupTeams.map((teamName, index) => ({
    position: index + 1,
    team: {
      name: teamName,
      shortName: teamName.substring(0, 3).toUpperCase(),
      crest: "" 
    },
    playedGames: 0,
    won: 0,
    draw: 0,
    lost: 0,
    points: 0,
    goalDifference: 0
  }));

  standings.push({
    stage: "GROUP_STAGE",
    type: "TOTAL",
    group: groupName,
    table: table
  });
}

// Generate Fixtures (Matches)
const matches = [];
let matchId = 1000;

// Hardcode Group D exactly as user's screenshot
const groupDFixtures = [
  { id: matchId++, utcDate: "2026-06-13T04:00:00Z", status: "TIMED", homeTeam: { name: "USA", shortName: "ABD" }, awayTeam: { name: "Paraguay", shortName: "PAR" }, venue: "Los Angeles Stadyumu", group: "D Grubu" },
  { id: matchId++, utcDate: "2026-06-14T07:00:00Z", status: "TIMED", homeTeam: { name: "Australia", shortName: "AVU" }, awayTeam: { name: "Turkey", shortName: "TÜR" }, venue: "Vancouver Stadyumu", group: "D Grubu" },
  { id: matchId++, utcDate: "2026-06-20T01:00:00Z", status: "TIMED", homeTeam: { name: "USA", shortName: "ABD" }, awayTeam: { name: "Australia", shortName: "AVU" }, venue: "Seattle Stadyumu", group: "D Grubu" },
  { id: matchId++, utcDate: "2026-06-20T06:00:00Z", status: "TIMED", homeTeam: { name: "Turkey", shortName: "TÜR" }, awayTeam: { name: "Paraguay", shortName: "PAR" }, venue: "San Francisco Stadyumu", group: "D Grubu" },
  { id: matchId++, utcDate: "2026-06-26T05:00:00Z", status: "TIMED", homeTeam: { name: "Paraguay", shortName: "PAR" }, awayTeam: { name: "Australia", shortName: "AVU" }, venue: "San Francisco Stadyumu", group: "D Grubu" },
  { id: matchId++, utcDate: "2026-06-26T05:00:00Z", status: "TIMED", homeTeam: { name: "Turkey", shortName: "TÜR" }, awayTeam: { name: "USA", shortName: "ABD" }, venue: "Los Angeles Stadyumu", group: "D Grubu" },
];
matches.push(...groupDFixtures);

// Add some generic matches for Group A to fill up the list
const groupAFixtures = [
  { id: matchId++, utcDate: "2026-06-11T19:00:00Z", status: "TIMED", homeTeam: { name: "Brazil", shortName: "BRA" }, awayTeam: { name: "Scotland", shortName: "SCO" }, venue: "Azteca Stadyumu", group: "A Grubu" },
  { id: matchId++, utcDate: "2026-06-12T15:00:00Z", status: "TIMED", homeTeam: { name: "Ecuador", shortName: "ECU" }, awayTeam: { name: "Japan", shortName: "JPN" }, venue: "Guadalajara Stadyumu", group: "A Grubu" },
];
matches.push(...groupAFixtures);

const workerCode = `const CORS_HEADERS = {
  "Access-Control-Allow-Origin": "*",
  "Access-Control-Allow-Methods": "GET, HEAD, POST, OPTIONS",
  "Access-Control-Allow-Headers": "Content-Type",
};

const MOCK_FIXTURES = {
  "matches": ${JSON.stringify(matches, null, 2)}
};

const MOCK_STANDINGS = {
  "standings": ${JSON.stringify(standings, null, 2)}
};

export default {
  async fetch(request, env, ctx) {
    if (request.method === "OPTIONS") {
      return new Response(null, { headers: CORS_HEADERS });
    }

    const url = new URL(request.url);

    if (url.pathname === "/") {
      return new Response("World Cup 2026 Ultimate Mock API Worker is running!", {
        status: 200,
        headers: { ...CORS_HEADERS, "Content-Type": "text/plain" },
      });
    }

    if (url.pathname === "/fixtures") {
      return new Response(JSON.stringify(MOCK_FIXTURES), {
        status: 200,
        headers: { ...CORS_HEADERS, "Content-Type": "application/json" }
      });
    }

    if (url.pathname === "/standings") {
      return new Response(JSON.stringify(MOCK_STANDINGS), {
        status: 200,
        headers: { ...CORS_HEADERS, "Content-Type": "application/json" }
      });
    }

    return new Response("Not Found", { 
      status: 404, 
      headers: { ...CORS_HEADERS, "Content-Type": "text/plain" } 
    });
  }
};
`;

fs.writeFileSync('cloudflare/worker.js', workerCode);
console.log('Worker generated successfully.');

File: generate_mock.py
import json
import datetime

with open('worldcup_2022.json', 'r') as f:
    data = json.load(f)

api_fixtures = []
standings_dict = {}

# Compute standings
for match in data.get('matches', []):
    group = match.get('group')
    if not group:
        group = 'Knockout'
    t1 = match.get('team1')
    t2 = match.get('team2')
    
    if group != 'Knockout':
        if group not in standings_dict:
            standings_dict[group] = {}
        for t in [t1, t2]:
            if t not in standings_dict[group]:
                standings_dict[group][t] = {
                    'team': {'name': t, 'shortName': t[:3].upper()},
                    'points': 0, 'playedGames': 0, 'won': 0, 'draw': 0, 'lost': 0, 'goalsFor': 0, 'goalsAgainst': 0, 'goalDifference': 0
                }
        
        score_ft = match.get('score', {}).get('ft', [])
        if score_ft and len(score_ft) == 2 and isinstance(score_ft[0], int):
            s1, s2 = score_ft
            st1 = standings_dict[group][t1]
            st2 = standings_dict[group][t2]
            st1['playedGames'] += 1
            st2['playedGames'] += 1
            st1['goalsFor'] += s1
            st1['goalsAgainst'] += s2
            st2['goalsFor'] += s2
            st2['goalsAgainst'] += s1
            st1['goalDifference'] += (s1 - s2)
            st2['goalDifference'] += (s2 - s1)
            
            if s1 > s2:
                st1['won'] += 1
                st1['points'] += 3
                st2['lost'] += 1
            elif s2 > s1:
                st2['won'] += 1
                st2['points'] += 3
                st1['lost'] += 1
            else:
                st1['draw'] += 1
                st2['draw'] += 1
                st1['points'] += 1
                st2['points'] += 1

    # Format for API-Sports
    date_str = match.get('date')
    time_str = match.get('time')
    
    dt = datetime.datetime.strptime(f"{date_str} {time_str}", "%Y-%m-%d %H:%M")
    delta = datetime.datetime(2026, 6, 11) - datetime.datetime(2022, 11, 20)
    new_dt = dt + delta
    
    utc_date = new_dt.isoformat() + "Z"
    
    api_fixtures.append({
        'utcDate': utc_date,
        'homeTeam': {'name': t1, 'shortName': t1[:3].upper()},
        'awayTeam': {'name': t2, 'shortName': t2[:3].upper()},
        'score': match.get('score'),
        'group': group.replace("Group ", "GROUP_") if group else "Knockout"
    })

# Format standings to match API
api_standings = []
for group_name, teams in standings_dict.items():
    sorted_teams = sorted(teams.values(), key=lambda x: (x['points'], x['goalDifference'], x['goalsFor']), reverse=True)
    for i, t in enumerate(sorted_teams):
        t['position'] = i + 1
    # API-Sports format: list of tables, each table is a group
    api_standings.append({
        'group': group_name.replace("Group ", "GROUP_"),
        'type': 'TOTAL',
        'table': sorted_teams
    })

with open('lib/data/mock_data.dart', 'w') as f:
    f.write('class MockData {\n')
    f.write('  static const List<dynamic> fixtures = ')
    f.write(json.dumps(api_fixtures, indent=2))
    f.write(';\n\n')
    f.write('  static const List<dynamic> standings = ')
    f.write(json.dumps(api_standings, indent=2))
    f.write(';\n}\n')

print("Mock data generated in lib/data/mock_data.dart")
